### PR TITLE
Add dependabot support

### PR DESCRIPTION
**Pull request by EnricoMi**
How do you think about dependabot support?

https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically

It creates PRs that upgrade dependencies, which can be merged with a single PR comment:
https://github.com/EnricoMi/publish-unit-test-result-action/pull/112

The changes to `ci.yml` and `ci-fork.yml` are required because dependabot PRs run like fork PRs isolated from your repository (no access to secrets) because new dependencies can pull in malicious code through those dependencies. Test results cannot be posted to the pull request from the CI workflow that runs the depdendabot PR.

### CODE REVIEW

**Comment by jgiannuzzi** [comment]
I'm personally very much in favour of it.

Could you please also enable support for `docker` and `npm`?

**Comment by EnricoMi** [comment]
Sure, is weekly fine?

**Comment by jgiannuzzi** [comment]
Sounds good to me.

@mariannovotny what are your thoughts?

**Comment by EnricoMi** [comment]
I am not sure if `directory: /` is sufficient for those `package-ecosystem`s that don't have their config file in the root directory. Meaning: does dependabot look into subdirectories or not. Would be quite tedious to maintain the exhaustive list of directories that contain `pom.xml`, `Dockerfile` or `package.json`.

Once merged, GitHub UI is quite verbose which files are scanned: https://github.com/EnricoMi/publish-unit-test-result-action/network/updates

File: .github/dependabot.yml
Comment (jgiannuzzi) [inline .github/dependabot.yml:8]: all of our Dockerfiles are in `/deployment/docker`

File: .github/dependabot.yml
Comment (jgiannuzzi) [inline .github/dependabot.yml:16]: our npm packages are in `/config-editor/config-editor-ui/`

File: .github/workflows/ci.yml
Comment (jgiannuzzi) [inline .github/workflows/ci.yml:39]: Not related to this PR, but I wonder why we have `|| github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id` here?
If this is a pull request from a fork, then we won't be able to publish the unit test results anyway, right?

Comment (EnricoMi) [reply .github/workflows/ci.yml:39]: You are right, that was a last-minute copy-paste error from the next step. To be fixed in #71.

**Comment by EnricoMi** [comment]
I have adjusted the docker and npm path. I think maven will require listing all paths. I would prefer to go ahead and merge into master as is, once @mariannovotny is happy with this. When dependabot then indicates only root pom.xml is monitored, then I will add an approach that keeps the dependabot.yml file in sync with project changes. Stay tuned.

**Comment by mariannovotny** [comment]
@EnricoMi Thanks for this PR. I hope dependabot will update package-lok.json and is able to handle multimodule maven project. But we will see.
Just to double check - The PR will ben not auto merged and needs to be approved by codeowners?

**Comment by EnricoMi** [comment]
> Just to double check

Yes. no auto-merge, full control to reviewers. Fine-tuning (e.g. what not to ignore) once we see upgrade PRs coming in.

**Review by mariannovotny — APPROVED** [review]

**Comment by EnricoMi** [comment]
Dependabot picked up these package files:
- .github/workflow/ci.yaml
- .github/workflow/ci-fork.yaml
- config-editor/config-editor-ui/package.json
- config-editor/config-editor-ui/package-lock.json
- deployment/docker/Dockerfile.config-editor-ui
- deployment/docker/Dockerfile.java
- deployment/docker/Dockerfile.storm
- pom.xml plus 21 pom.xml in sub-directories

grafik

**Comment by EnricoMi** [comment]
Dependabot created 16 PRs upgrading dependencies individually. From the CI pipeline, you can see if the dependency breaks things. With Depandabot commands listed in the PR description, you can tell Dependabot what to do.

grafik

After one dependency is merged into master, you should rebase the next dependency and proceed.

**Comment by EnricoMi** [comment]
Be aware, dependabot would have liked to open more pull requests:

grafik

**Pull request merged**
